modules/database-commons/src/main/java/org/testcontainers/ext/ScriptUtils.java

```diff
@@ -163,19 +163,19 @@ public static void splitSqlScript(String resource, String script, String separat
 			}
 			final boolean inComment = inLineComment || inBlockComment;
 
-			if (!inLiteral && !inComment && containsSubstringAtOffset(lowerCaseScriptContent, "BEGIN", i)) {
+			if (!inLiteral && !inComment && containsKeywordsAtOffset(lowerCaseScriptContent, "BEGIN", i, separator, commentPrefix, blockCommentStartDelimiter)) {
 				compoundStatementDepth++;
 			}
-			if (!inLiteral && !inComment && containsSubstringAtOffset(lowerCaseScriptContent, "END", i)) {
+			if (!inLiteral && !inComment && containsKeywordsAtOffset(lowerCaseScriptContent, "END", i, separator, commentPrefix, blockCommentStartDelimiter)) {
 				compoundStatementDepth--;
 			}
 			final boolean inCompoundStatement = compoundStatementDepth != 0;
 
 			if (!inLiteral && !inCompoundStatement) {
 				if (script.startsWith(separator, i)) {
 					// we've reached the end of the current statement
-					if (sb.length() > 0) {
-						statements.add(sb.toString());
+					if (StringUtils.isNotBlank(sb.toString())) {
```

Comment:
we should at least extract `sb.toString()`, since we do it again inside the `if` block, and, given that we're inside a loop, that will affect the performance

Reply:
Ah good catch. I did this for consistency but neglected the performance. Will change this evening.

Reply:
Updated to be both more performant and to reduce duplication